### ISSUE TITLE
Make the headings & subheadings more distinct

### ISSUE THREAD

**Issue by johnmcollier** [eclipse/codewind#512]
In our docs, I've found that the headings and subheadings for docs are hard to tell apart (and are pretty much the same size). We should make them more distinct (probably make the subheadings smaller) to be easier for users to read

Current Examples:
![Screen Shot 2019-09-10 at 11 09 04 AM](https://user-images.githubusercontent.com/6880023/64626266-91cd3280-d3bb-11e9-8f6e-a47a91c8669e.png)

![Screen Shot 2019-09-10 at 11 09 31 AM](https://user-images.githubusercontent.com/6880023/64626280-9691e680-d3bb-11e9-8e42-1b9558be5cbe.png)

**Comment by sishida** [comment]
@kimholmes Could design help improve our headings to make them more distinct with different sizes/colors/letter thickness/etc.?

Also tagging @travis1111 for any upcoming implementation after we decide on designs for the headings.

If either of you would like to be assigned to this issue, please leave a comment. (GitHub won't allow me to assign you without you first leaving a comment.)

**Comment by MelHopper** [comment]
@kimholmes the other item we didn't have time to discuss - the best-practice design/style for topics that I've ever come across - the styling of this webpage made following this tutorial incredibly easy:

https://docs.docker.com/get-started/part2/

(see tabs on the top of the page for other parts within this tutorial)

The webpage uses:
- mixture of header sizes
- marks on text to indicate 'aside' content blocks
- colours to differentiate 
- blue colours with grey background for code snippets and blocks vs. black font with white background for general content
'continue to part x' buttons rather than just text

Do need to bear in mind accessibility here when styling.

**Comment by kimholmes** [comment]
Thanks. I will roll this into the overall revamp we're doing for the Docs page. I've created a new issue in the design repo here:
https://github.ibm.com/dev-ex/icap-design/issues/94

**Comment by sishida** [comment]
@kimholmes Could we work on the headings/subheadings design separately from the rest of the docs redesign? I don't think any other pieces of the redesign are dependent on the headings designs, so it'd be good if we could work on the headings sooner. :)

**Comment by johnmcollier** [comment]
Any updates on this?

**Comment by thisguy-1** [comment]
@sishida  So I can be assigned

**Comment by sishida** [comment]
@travis1111 What if we changed the Codewind website doc headings so that they match the headings of basic Markdown? https://www.markdownguide.org/basic-syntax/

I checked the current Codewind website, and it seems that the current headings don't quite match these Markdown headings. It looked like there might have been bold, underlining, and a different color. I'm wondering if it would be better to just omit all of these extra elements and just start with size differences?

**Comment by rajivnathan** [comment]
Any updates on this? It makes the docs rather difficult to navigate and the docs are the first place someone new to the product will look.

**Comment by sghung** [comment]
I will followup with Kim on this to see if Sarah's suggestion is suitable

**Comment by kimholmes** [comment]
I assigned this to @thisguy-1 before the holiday.
Re: Sarah's suggestion. It would make more sense to me if we used the Carbon Guidelines:
https://www.carbondesignsystem.com/guidelines/typography/overview

Those are based in CSS, and that's you're using on the site, correct?

**Comment by travis1111** [comment]
So you suggestion is to use carbon scale?

https://www.carbondesignsystem.com/guidelines/typography/overview/#scale

**Comment by sishida** [comment]
Documentation for Codewind is written in Markdown, which is why I suggested  https://www.markdownguide.org/basic-syntax/. Making the heading match the Markdown defaults would be like a "factory reset" of our headings. :)

Of course, if design wants to design new headings, then I am all for the Carbon guidelines!

Another thing to consider is limiting our use of headings. For example, if we decide we only want to use heading levels 1, 2, and 3, we only really need designs for three things...

**Comment by kimholmes** [comment]
Ok @sishida If it's already written in Markdown (although, that seems a little strange since it's a website), then I think it's fine. Whatever is easiest. Is it also not possible to look at the code for Open Liberty or another site and see what they have?

**Comment by sishida** [comment]
@kimholmes, The doc files are Markdown files: https://github.com/eclipse/codewind-docs/tree/master/docs/_documentations

The website itself is separate, but the Markdown in these files has an impact on how that content appears on the website.

For example, Markdown uses `#` characters to determine heading size. The more of the `#` characters, the smaller the heading appears.

**Comment by travis1111** [comment]
Thanks all!

We will use  https://www.markdownguide.org/basic-syntax/

**Comment by travis1111** [comment]
Was playing around with different fonts/sizes of the headings:

1. headings on https://www.markdownguide.org/basic-syntax/

![image](https://user-images.githubusercontent.com/35569780/72531765-2818d580-3840-11ea-93e1-0a0aae6b43a6.png)

2. Heading on codewind landing page (font-family: 'IBM Plex Sans', sans-serif;)

![image](https://user-images.githubusercontent.com/35569780/72531817-45e63a80-3840-11ea-9e81-65368afd1312.png)

3. Heading on codewind landing page (same font family as markdownguide.org)

![image](https://user-images.githubusercontent.com/35569780/72531933-7a59f680-3840-11ea-803a-b57236965f5a.png)

Basically 1 and 3 are the same.  2 is different is because we use different font family.

For `<strong>` tag I can make it bolder.

**Comment by sishida** [comment]
@travis1111 I think these look good! The ID team is trying to decide on which heading sizes we will use as we go forward, so users who look at our content should only see maybe three different heading sizes at most throughout the "docs" tab on the website. (We do appreciate the option to use all of them, though! :) )

@kimholmes Do these look good to go to you?

**Comment by kimholmes** [comment]
These look fine, but yes, you only really need about 3 sizes.

**Comment by travis1111** [comment]
#2 is what we have currently

**Comment by sishida** [comment]
@travis1111 Thanks for all your work on this! Just so I ensure I understand correctly...
1️⃣ If the team wants to make the titles bold, we just use the `<strong>` tag, correct?
2️⃣ When you say #2 is what we have currently, that is what's currently been merged to master? The 0.8.0 release branch was already branched off earlier this week, so if we want the updated headers to go out with 0.8.0, we also need to add the changes to that release branch. I am personally OK with it going out later with 0.9.0 if it'll be time consuming or difficult to add it to the release branch.
Thanks!

**Comment by travis1111** [comment]
1. No. The `<strong>` tag I mentioned was to respond to this comment:

https://github.com/eclipse/codewind/issues/512#issuecomment-530967273

2. No. we are currently using markdown defaults.  No changes have been made  (except I will make `<strong>` bolder.  But this is irrelevant to the headings.

**Comment by travis1111** [comment]
@johnmcollier please take a look of https://github.com/eclipse/codewind/issues/512#issuecomment-575170584

thanks!

**Comment by johnmcollier** [comment]
IMO, #1 and #3 (keeping in mind #2 is what we have right now), Heading levels 2 and 3 are still too similar and hard to distinguish. What about @jgwest's suggestion on adding an underline to `h2`?

**Comment by johnmcollier** [comment]
If you view the docs' markdown files in Github, all 3 heading sizes are nicely distinct, and having underlines under them helps
<img width="903" alt="Screen Shot 2020-01-16 at 5 28 50 PM" src="https://user-images.githubusercontent.com/6880023/72568292-cf6c2b80-3885-11ea-99e1-543850d9977d.png">

**Comment by travis1111** [comment]
what about this

![image](https://user-images.githubusercontent.com/35569780/72571542-e6168080-388d-11ea-8585-844a145f1774.png)

**Comment by sishida** [comment]
I like how the underlining helps separate the headings from the body text. I think all sizes of headings should come with underlining. 👍

**Comment by travis1111** [comment]
OK, I will add to all headings, if we think it is to distracting we can remove some of them.

**Comment by kimholmes** [comment]
Thank you @travis1111 and @sishida Using the rules does create a nice separation between the blocks of content. It might be good to use those only after the main headers but leave them off below the subheaders. I actually like to use those to differentiate blocks of content.

This isn't in regard to the fonts, I think those are looking great, but one thing that really stands out to me in that screenshot is the padding between the sections. I think it needs quite a bit more padding.
Example:
![image](https://user-images.githubusercontent.com/21316329/72640568-20d3f380-3936-11ea-9d88-e8dd517b5e1c.png)

**Comment by travis1111** [comment]
@kimholmes Thanks.  If I compare the screen shot you have with the example I pasted, you removed underline on H2, and added more top padding on H1.  Is that correct?  Are you suggesting we only add underline on H1?

Thanks

**Comment by malincoln** [comment]
@johnmcollier pls verify and close.  Thanks
